plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -38,6 +43,50 @@
 		ResourceKind.ImageStream, 
 		ResourceKind.Route
 	};
+	
+	private Map<IProject, List<ResourceGrouping>> groupMap = new HashMap<IProject, List<ResourceGrouping>>();
+	
+	@Override
+	protected void handleConnectionChanged(IConnection connection, String property, Object oldValue, Object newValue) {
+		if(!(connection instanceof Connection))return;
+		if(ConnectionProperties.PROPERTY_RESOURCE.equals(property)) {
+			if(oldValue == null) {
+				//add
+			}else if(newValue == null) {
+				//delete
+				IResource resource = (IResource)oldValue;
+				refreshGrouping(groupMap.get(resource.getProject()), resource.getKind());
+			}else {
+				//change
```

Comment:
updated though im wondering if anything will happen since the refresh viewer takes an object and what we really need is an update and a way to pass the properties that updated.